homeassistant/components/youtube/entity.py

```diff
@@ -21,20 +18,26 @@ def __init__(
         self,
         coordinator: YouTubeDataUpdateCoordinator,
         description: EntityDescription,
-        channel: dict[str, Any],
+        channel_id: str,
     ) -> None:
-        """Initialize a Google Mail entity."""
+        """Initialize a YouTube entity."""
         super().__init__(coordinator)
         self.entity_description = description
         self._attr_unique_id = (
-            f"{coordinator.config_entry.entry_id}_{channel[ATTR_ID]}_{description.key}"
-        )
-        self._attr_device_info = DeviceInfo(
-            entry_type=DeviceEntryType.SERVICE,
-            identifiers={
-                (DOMAIN, f"{coordinator.config_entry.entry_id}_{channel[ATTR_ID]}")
-            },
-            manufacturer=MANUFACTURER,
-            name=channel[ATTR_TITLE],
+            f"{coordinator.config_entry.entry_id}_{channel_id}_{description.key}"
         )
-        self._channel = channel
+        self._channel_id = channel_id
+        if self.available:
```

Comment:
This is not good. this means that in some cases the entity is not tied to the device. 

Why would it not be available on setup? If the data could not be fetched, I would assume that the setup raised ConfigEntryNotReady

Reply:
Whoops, was still left behind from another thing I was looking into